src/backend/dx11/shaders/copy.hlsl

```diff
@@ -552,10 +599,28 @@ void cs_copy_buffer_image2d_r8(uint3 dispatch_thread_id : SV_DispatchThreadID) {
     uint src_idx = GetBufferSrc8(dispatch_thread_id);
     uint4 data = Uint32ToUint8x4(BufferCopySrc.Load(src_idx));
 
-    Image2CopyDstR[dst_idx + uint3(0, 0, 0)] = data.x;
-    Image2CopyDstR[dst_idx + uint3(1, 0, 0)] = data.y;
-    Image2CopyDstR[dst_idx + uint3(2, 0, 0)] = data.z;
-    Image2CopyDstR[dst_idx + uint3(3, 0, 0)] = data.w;
+    uint remaining_x = bounds.x - dst_idx.x;
+
+    switch (remaining_x) {
+        default:
+            Image2CopyDstR[dst_idx + uint3(0, 0, 0)] = data.x;
+            Image2CopyDstR[dst_idx + uint3(1, 0, 0)] = data.y;
```

Comment:
would it be simpler to avoid repeating ourselves and just adding a condition for each of the assignments?
```cpp
if (remaining_x > 0) {
  Image2CopyDstR[dst_idx + uint3(0, 0, 0)] = data.x;
}
if (remaining_x > 1) {
  Image2CopyDstR[dst_idx + uint3(1, 0, 0)] = data.y;
}
...
```
We could even just loop there

Reply:
I don't think we can loop due to data being in a uvec4 (now yes we could put it in an array, but meh). If statements though should actually be a bit more coherent, so it should be a win.

Reply:
you should be able to index `uvec4` as an array, no need to put it out